### PR TITLE
converting search to a more rounded search bar

### PR DESCRIPTION
**Pull request by d88naimi**
Rounded search bar is now able to be turned on with passing a props `roundedSearch: true`
Screen Shot 2021-07-18 at 6 03 22 AM

Docs 
Screen Shot 2021-07-18 at 6 13 56 AM

### CODE REVIEW

File: core/src/InputIcon/InputIcon.less [hunk 1]
```diff
@@ -9,4 +9,7 @@
     font-size: @svg-icon;
     margin: 0;
   }
+  &.RoundedSearchIcon {
```
Comment (ardeay) [inline core/src/InputIcon/InputIcon.less:12]: Also drop the box shadow here

Is it possible to override from the search component LESS?

File: core/src/Search/Search.js [hunk 2]
```diff
@@ -60,10 +60,16 @@ class Search extends Component {
...
           type="search"
-          className={styles.Input}
+          className={
+            this.props.roundedSearch ? styles.RoundedSearch : styles.Input
+          }
           onChange={this.onChange}
+          roundedSearch={this.props.roundedSearch || null}
         />
-        <InputIcon onClick={this.onSubmit}>
+        <InputIcon
```
Comment (ardeay) [inline core/src/Search/Search.js:69]: Is it possible to pass className={styles.roundedEdge} which comes from the search component less file

**Review by ardeay — CHANGES_REQUESTED** [review]
The button input 
```
    box-shadow: none;
    background: white;
    border: #??? solid 1px; /** have this match the input border */
    border-left: none;
    color: @zesty-gray;
```

input 
```
border-right: none;
````
It would look something like this (colors are off because i did it in browser)
image

**Comment by d88naimi** [comment]
Version 2.0 roundedEdge= "true" 
all styles are controlled in search component and removed from inputIcon.js/less
Screen Shot 2021-07-19 at 11 50 52 AM

**Comment by shrunyan** [comment]
Why are we making this an option? 

The design system should provide consistent solutions for common UI patterns/problems. Why would our interfaces need both rounded and less rounded inputs? I feel we should pick one or the other. This way users can have a single learned behavior on what search inputs look like. Allowing for lower cognitive load and quicker scanning to find that feature (i.e. don't make me think)

**Comment by d88naimi** [comment]
Screen Shot 2021-07-19 at 3 56 28 PM

Using Content View image as the example. I connect the rectangle inputs with a textfield/dropdown. By adding the rounded search option it allows the user to differentiate the search input from the other two fields above while having a more aesthetically view. 

Currently I feel there is more cognitive load, why do I have a global search and content search which one is better? By having the rounded search inside the content view, I can deduce that the rounded search is more accurate in searching the Content view.

CC: @ardeay

**Pull request merged**